### DISCUSSION TITLE
[Feature] Import shared immich link to my immich

### DISCUSSION THREAD

**Discussion by palitu** [immich-app/immich#2730] [Feature Request]
### The feature

Last week a friend sent us a link to one of their immich albums, which was a shared holiday.

**I would like to be able to copy the files into my immich instance so that i have my own copy of the files.**

**CORE FEATURES**
- when sharing a link, provide a way to not allow direct copying into another immich instance
- If link is opened on a mobile, associate immich type links with the app, and open the shared images with the immich mobile app (otherwise, provide a method to copy an immich share into the immich app/website)
- Provide an option to copy all in the share to local immich
- provide a method to individually select one or more images to copy to the local instance

**OPTIONAL FEATURES**
- add meta data to the image to store where the images came from
- auto-create an album that has all the copied images in it

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by mirisbowring** [comment]
Do you mean they have their dedicated instance and you would like to import to your instance, or are they users on your instance?

First one sounds a bit like a Fediverse Feature which i would love also 😄

**Comment by axgd-code** [comment]
I find this proposal crucial for my family to transition to Immich. Indeed, we all have our own libraries, but we would like to have shared albums in common. It is easier to see in a dedicated tab the albums shared by other instances. I believe that the partial implementation of ActivityPub would allow for a more standard exchange format for other features in the future: https://www.w3.org/TR/activitypub/#shares.

I envision being able to have a toggle switch in my library: 'Albums shared with me'. The displayed albums would be those shared on the same instance or distant instances. 
To add an album in my instance, I imagine two approaches: creating a virtual album where 'we retrieve the photos from the other instance every time they are loaded', or one could choose to download the images as soon as a photo is added.

**Comment by Qhilm** [comment]
In the Immich app on iOS, I hit "download" for a selected picture (long press on thumbnail till blue checkmark appears) part of an album shared with me by another immich user on the same instance, and it adds the picture to my timeline. All other pictures in the album remain hidden from my timeline. Bit confusing to be honest, this is not what I expect when hitting something labeled "download". I guess it's inpired from the google photo icon to import into own library (a cloud with a down arrow), but it really should be called "import to my library" rather, or "import to my photos".

The only issue I found was, it takes time for the image to appear in my timeline, sometimes minutes and I had to restart the immich app once to see the picture in the timeline.

I couldn't find a way to "import entire shared album into own library/timeline" though.
